python/cudf/cudf/core/index.py

```diff
@@ -2357,6 +2357,32 @@ def is_leap_year(self):
         res = is_leap_year(self._values).fillna(False)
         return cupy.asarray(res)
 
+    @property
+    def quarter(self):
+        """
+        Integer indicator for which quarter of the year the date belongs in.
+
+        There are 4 quarters in a year. With the first quarter being from
+        January - March, second quarter being April - June, third quarter
+        being July - September and fourth quarter being October - December.
+
+        Returns
+        -------
+        Int64Index
+        Integer indicating which quarter the date belongs to.
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> import cudf
+        >>> pIndex = pd.DatetimeIndex(["2020-05-31 08:00:00",
+        ...    "1999-12-31 18:40:00"])
+        >>> cudf.from_pandas(pIndex).quarter
+        Int64Index([2, 4], dtype='int64')
```

Comment:
Just one more nitpick: while we can, let's try to limit the example construction to using only cudf. This makes the example cleaner too.

```suggestion
        >>> import cudf
        >>> gIndex = cudf.DatetimeIndex(["2020-05-31 08:00:00",
        ...    "1999-12-31 18:40:00"])
        >>> gIndex.quarter
        Int64Index([2, 4], dtype='int64')
```